server/service/users_test.go

```diff
@@ -671,29 +684,29 @@ func testUsersChangePassword(t *testing.T, ds *mysql.Datastore) {
 	}{
 		{ // all good
 			user:        users["admin1@example.com"],
-			oldPassword: "foobarbaz1234!",
-			newPassword: "12345cat!",
+			oldPassword: test.GoodPassword,
+			newPassword: test.GoodPassword2,
 		},
 		{ // prevent password reuse
 			user:        users["admin1@example.com"],
-			oldPassword: "12345cat!",
-			newPassword: "foobarbaz1234!",
+			oldPassword: test.GoodPassword2,
+			newPassword: test.GoodPassword,
```

Comment:
Regarding the `wantErr:     fleet.NewInvalidArgumentError("new_password", "cannot reuse old password")`:
Is this because we first validate the new password? Shouldn't we first validate if the `oldPassword` is valid?

Reply:
@lucasmrod, good question. It looks like the old `ChangePassword` method did not include separate verification of the old password included in the request. I presume this was because there is a auth check happening [here](https://github.com/fleetdm/fleet/blob/main/server/service/users.go#L496) that would have required the old user password. Do you think that is sufficient?

Reply:
```go
func (svc *Service) ChangePassword(ctx context.Context, oldPass, newPass string) error {
        // [...]
        if err := vc.User.ValidatePassword(newPass); err == nil {
		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("new_password", "cannot reuse old password"))
	}
	if err := vc.User.ValidatePassword(oldPass); err != nil {
		return ctxerr.Wrap(ctx, fleet.NewInvalidArgumentError("old_password", "old password does not match"))
	}

	if err := svc.setNewPassword(ctx, vc.User, newPass); err != nil {
		return ctxerr.Wrap(ctx, err, "setting new password")
	}
	return nil
}
```

Ah found it, we should swap the `ValidatePassword` calls 👆 . But we don't need to do this now. I'll create a separate issue/PR.